manila/share/manager.py

```diff
@@ -5642,6 +5681,20 @@ def _server_migration_complete_driver(self, context, source_share_server,
         dest_sn = self.db.share_network_get(context, dest_sn_id)
         dest_sns = self.db.share_network_subnet_get(context, dest_sns_id)
 
+        # SAPCC: Network are extended to the destination host on previous
+        # (migration_start) step, i.e. port bindings are created on destination
+        # host using the old ports. The network allocations will be cut over on
+        # this (migration_complete) step, i.e. port bindings on destination
+        # host will be activated and bindings on source host will be deleted.
+        # Since network allocations has been taken care of in the cutover,
+        # driver should not touch them. Therefore source_share_server and
```

Comment:
The last sentence is not clear to me. Aren't `migration_reused_network_allocations` and `share_server_was_reused` conditions later handling that?

Reply:
I mean port bindings cut over are handled in manager, driver does not need to take extra care of them. The driver function `share_server_migration_complete` takes `source_share_server`, `dest_share_server`, and `new_network_allocations` as arguments. Here the `source_share_server` and `dest_share_server` are not updated after the cut-over, so the driver has no knowledge of any port-binding changes and therefore do nothing.

Reply:
the later block `if not migration_reused_network_allocations:` does not seem to have any real effect. The block is executed only when `migration_reused_network_allocations` is False. In this case, `server_to_get_allocations` is `dest_share_server`. But then in this block, db is updated with the value `{'share_server_id': dest_share_server[
id]}`. Aren't the allocations are already owned by the dest_share_server?

Reply:
> driver does not need to take extra care

Thanks for the confirmation - that interpretation seemed the most preferred to me